src/main/java/tutorlink/lists/GradeList.java

```diff
@@ -53,12 +57,40 @@ public GradeList findGrade(String matricNumber, String componentDescription) thr
         filteredList.gradeArrayList = gradeArrayList
                 .stream()
                 .filter(grade -> grade.getStudent().getMatricNumber().equals(matricNumber.toUpperCase())
-                && grade.getComponent().getName().equals(componentDescription.toLowerCase()))
+                        && grade.getComponent().getName().equals(componentDescription.toLowerCase()))
```

Comment:
I like that you are matching both the name and the matriculation number for calculating grade. Is there a reason why this is preferred instead of just using the matriculation number?